apps/nextjs/src/stores/chatStore/actionFunctions/handleQueueUserAction.ts

```diff
@@ -0,0 +1,8 @@
+import type { ChatStore } from "..";
+
+export const handleQueueUserAction =
+  (set: (partial: Partial<ChatStore>) => void, get: () => ChatStore) =>
+  async (action: string) => {
+    set({ queuedUserAction: action });
+    await get().executeQueuedAction();
```

Comment:
Just checking that you want to execute the queued action at the same time as queueing it. I would expect this to queue it first, and then later execute when the streaming finishes